.github/workflows/js.yml

```diff
@@ -0,0 +1,66 @@
+name: CheckJS
+
+on:
+  # Run on pushes to select branches and on all pull requests.
+  push:
+    branches:
+      - master
+      - trunk
+      - 'release/**'
+      - 'hotfix/[0-9]+.[0-9]+*'
+      - 'feature/**'
+  pull_request:
+  # Allow manually triggering the workflow.
+  workflow_dispatch:
+
+# Cancels all previous workflow runs for the same branch that have not yet completed.
+concurrency:
+  # The concurrency group contains the workflow name and the branch name.
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  yarn-lint-test:
+    runs-on: ubuntu-latest
+
+    name: "CheckJS"
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      # The ubuntu images come with Node, npm and yarn pre-installed.
+      # https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-Readme.md
+
+      # This action also handles the caching of the Yarn dependencies.
+      # https://github.com/actions/setup-node
+      - name: Set up node and enable caching of dependencies
+        uses: actions/setup-node@v3
+        with:
+          node-version: '14'
+          cache: 'yarn'
+
+      - name: Yarn install
+        run: yarn install
+
+      - name: Show debug info
+        run: |
+          npm --version
+          node --version
+          yarn --version
+          grunt --version
+
+      - name: Run Javascript lint
+        run: yarn lint
+
+      # Check out the premium config repo ahead of running the tests to prevent issues with permissions.
+      - name: Checkout premium configuration
+        uses: actions/checkout@v3
+        with:
+          repository: Yoast/YoastSEO.js-premium-configuration
```

Comment:
@diedexx I did consider doing that, but it would create a "maintaining duplicate logic in two very different places" situation, which is something I generally try to avoid, especially when it's not needed.

Why is not needed you ask ?
* The `Yoast/YoastSEO.js-premium-configuration` is a repo with limited history (~1000 commits), so I set the above `action/checkout` up to fetch the complete history via `fetch-depth: 0`.
    This is the same as would otherwise have been done with the clone in grunt, just more transparent and with the credentials sorted out.
* Then by the time the test runs which needs the repo, the repo is already in place, so there is no need to clone.
* And as the full history is available, the `grunt` script should be able to switch between the branches without problem as the branches are known locally and for local git operations, no credentials are needed.

Am I overlooking something ?

Reply:
Just looking at the transcript now.... hmm.. might be that it actually needs to have `origin/` in front of the env name to handle the check-out correctly:

```
yoastseo: $ grunt get-premium-configuration
yoastseo: Running "shell:get-current-branch" (shell) task
yoastseo: (HEAD
yoastseo: Running "shell:clone-premium-configuration" (shell) task
yoastseo: Done
yoastseo: Running "shell:checkout-premium-configuration" (shell) task
yoastseo: Already on 'develop'
yoastseo: Your branch is up to date with 'origin/develop'.
yoastseo: error: pathspec 'JRF/switch-to-ghactions-step4' did not match any file(s) known to git
yoastseo: Running "shell:pull-premium-configuration" (shell) task
```

https://github.com/Yoast/wordpress-seo/runs/5467094413?check_suite_focus=true#step:8:63

Reply:
Ah I completely misunderstood the fetch-depth! That makes a lot of sense.
I don't think the origin is required. The [JRF/switch-to-ghactions-step4 branch just doesn't exist](https://github.com/Yoast/YoastSEO.js-premium-configuration/branches/all?query=JRF%2Fswitch-to-ghactions-step4), so it falls back to `develop`.

Reply:
Correct. Maybe I should temporarily create a `JRF/...` branch in the premium config repo to verify that if it does exist, the branch switching works correctly ?

Reply:
Looks good 👍 
https://github.com/Yoast/wordpress-seo/runs/5507709199?check_suite_focus=true#step:8:64
```
yoastseo: Running "shell:clone-premium-configuration" (shell) task
yoastseo: Done
yoastseo: Running "shell:checkout-premium-configuration" (shell) task
yoastseo: Already on 'develop'
yoastseo: Your branch is up to date with 'origin/develop'.
yoastseo: Switched to a new branch 'JRF/switch-to-ghactions-step4'
yoastseo: branch 'JRF/switch-to-ghactions-step4' set up to track 'origin/JRF/switch-to-ghactions-step4'.
yoastseo: Running "shell:pull-premium-configuration" (shell) task
```